### PR TITLE
[REVIEW] Added in offset param host buffer readers in java.

### PR DESCRIPTION
**Pull request by revans2**
Some of the java APIs for host buffers were not consistent with java APIs.  They were missing an offset.  This fixes that.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 1]
```diff
@@ -254,38 +303,84 @@ public static Table readCSV(Schema schema, CSVOptions opts, HostMemoryBuffer buf
...
+      len = buffer.length - offset;
     }
     assert len > 0;
-    assert len <= buffer.length;
+    assert len <= buffer.length - offset;
+    assert offset >= 0 && offset < buffer.length;
     try (HostMemoryBuffer newBuf = HostMemoryBuffer.allocate(len)) {
       newBuf.setBytes(0, buffer, 0, len);
-      return readParquet(opts, newBuf, len);
+      return readParquet(opts, newBuf, 0, len);
```
Comment (abellina) [inline java/src/main/java/ai/rapids/cudf/Table.java]: I think this shouldn't be 0 and offset should be factored in.

Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/Table.java]: great catch.

Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/Table.java]: @abellina please take another look

Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/Table.java]: Should we add a test that passes an offset instead of 0 as an offset

Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/Table.java]: That is a lot harder to do with Parquet than it is with CSV.  So I would prefer not to, but if you really want one I can probably put one together.

Comment (abellina) [reply java/src/main/java/ai/rapids/cudf/Table.java]: Thanks @revans2. Would a simple test like allocate a slightly bigger byte[] than some Parquet buffer, then left pad it with something, s.t. we just have this read the Parquet payload at the offset?

Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/Table.java]: I am OK with it if you are OK

**Comment by revans2** [comment]
@abellina sorry could you take another look.  Had to fix a merge conflict in the CHANGELOG

**Pull request merged**